src/app/core/models/templates/tutor-moderation-issue-template.model.ts

```diff
@@ -1,21 +1,57 @@
 import { GithubIssue } from '../github/github-issue.model';
+import { IssueDispute } from '../issue-dispute.model';
+import { IssueDisputeSectionParser } from './sections/issue-dispute-section-parser.model';
 import { IssueDisputeSection } from './sections/issue-dispute-section.model';
 import { Section } from './sections/section.model';
 import { Header, Template } from './template.model';
 
+const { coroutine, everyCharUntil, many1, str, whitespace } = require('arcsecond');
+
 const tutorModerationIssueDescriptionHeaders = {
   description: new Header('Issue Description', 1),
   teamResponse: new Header("Team's Response", 1),
   disputes: new Header('Disputes', 1)
 };
 
+const DESCRIPTION_HEADER = '# Issue Description';
+const TEAM_RESPONSE_HEADER = "# Team's Response";
+const DISPUTES_HEADER = '# Disputes';
+
+export const TutorModerationIssueParser = coroutine(function* () {
+  yield str(DESCRIPTION_HEADER);
+  yield whitespace;
+  const description = yield everyCharUntil(str(TEAM_RESPONSE_HEADER));
+
+  yield str(TEAM_RESPONSE_HEADER);
+  yield whitespace;
+  const teamResponse = yield everyCharUntil(str(DISPUTES_HEADER));
+
+  // parse disputes
+  yield str(DISPUTES_HEADER);
+  yield whitespace;
+  const disputes = yield many1(IssueDisputeSectionParser);
+
+  // build array of IssueDisputes
+  const issueDisputes: IssueDispute[] = [];
+
+  for (const dispute of disputes) {
+    issueDisputes.push(new IssueDispute(dispute.title, dispute.description));
+  }
```

Comment:
Let's use array.map here

Reply:
i realise i can just create the new `IssueDispute` within the section parser itself so there's no need for this

Reply:
What i meant was something like `const issueDisputes = disputes.map(({title, desc}) => new IssueDispute(title, desc))`. I find that for simple array transformations, the functional paradigm is easier to read.